diff --git a/dist/changesets-assemble-release-plan.cjs.js b/dist/changesets-assemble-release-plan.cjs.js
index e07ba6e793021b6cfdec898afca517e293386ddb..ccaf0ad675108c14e6ba13e407202fdbe897019c 100644
--- a/dist/changesets-assemble-release-plan.cjs.js
+++ b/dist/changesets-assemble-release-plan.cjs.js
@@ -317,6 +317,9 @@ function shouldBumpMajor({
   preInfo,
   onlyUpdatePeerDependentsWhenOutOfRange
 }) {
+  if (depType === "peerDependencies") {
+    return false
+  }
   // we check if it is a peerDependency because if it is, our dependent bump type might need to be major.
   return depType === "peerDependencies" && nextRelease.type !== "none" && nextRelease.type !== "patch" && (
   // 1. If onlyUpdatePeerDependentsWhenOutOfRange set to true, bump major if the version is leaving the range.
diff --git a/dist/changesets-assemble-release-plan.esm.js b/dist/changesets-assemble-release-plan.esm.js
index ea2be567403c4ef94a65f3218ccb683cf5cb4bc1..9d787da1aeb80e629e55be0c58ef3aac98dad94a 100644
--- a/dist/changesets-assemble-release-plan.esm.js
+++ b/dist/changesets-assemble-release-plan.esm.js
@@ -306,6 +306,9 @@ function shouldBumpMajor({
   preInfo,
   onlyUpdatePeerDependentsWhenOutOfRange
 }) {
+  if (depType === "peerDependencies") {
+    return false
+  }
   // we check if it is a peerDependency because if it is, our dependent bump type might need to be major.
   return depType === "peerDependencies" && nextRelease.type !== "none" && nextRelease.type !== "patch" && (
   // 1. If onlyUpdatePeerDependentsWhenOutOfRange set to true, bump major if the version is leaving the range.
